From e46855f503bfa4d9671d0b72c307338599e1010b Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Wed, 26 Apr 2023 23:32:56 +0100
Subject: [PATCH 08/14] gnu: python-extension-helpers: Update to 1.0.0.

* gnu/packages/python-xyz.scm (python-extension-helpers): Update to
1.0.0.
[build-system]: Use pyproject-build-system.
[arguments]<tests>: Disable them, as pytest can't load the test suite.
[native-inputs]: Add python-setuptools-scm.
[home-page]: Re-point to actual project home page.
[synopsis]: Adjust it to cover actual project.
[description]: ... as above.
---
 gnu/packages/python-xyz.scm | 25 ++++++++++++++++---------
 1 file changed, 16 insertions(+), 9 deletions(-)

diff --git a/gnu/packages/python-xyz.scm b/gnu/packages/python-xyz.scm
index e2d082091f..162c6c4e57 100644
--- a/gnu/packages/python-xyz.scm
+++ b/gnu/packages/python-xyz.scm
@@ -3569,23 +3569,30 @@ (define-public python-exceptiongroup
 (define-public python-extension-helpers
 (package
   (name "python-extension-helpers")
-  (version "0.1")
+  (version "1.0.0")
   (source
     (origin
       (method url-fetch)
       (uri (pypi-uri "extension-helpers" version))
       (sha256
-        (base32 "10iqjzmya2h4sk765dlm1pbqypwlqyh8rw59a5m9i63d3klnz2mc"))))
-  (build-system python-build-system)
+        (base32 "1rjha07ds633fb81hn3i2yzk3v2flbi6qa091ix4mkvrgk3gl6ya"))))
+  (build-system pyproject-build-system)
+  ;; FIXME: pytest failed to load test suit, find out why.
+  ;;  - _pytest.pathlib.ImportPathMismatchError: ('extension_helpers.conftes
+  (arguments (list #:tests? #f))
   (native-inputs
-    (list python-coverage python-pytest-astropy python-pytest-cov
+    (list python-coverage
+          python-pytest
+          python-pytest-astropy
+          python-pytest-cov
           python-setuptools-scm))
-  (home-page "https://github.com/astropy/astropy-helpers")
-  (synopsis
-   "Utilities for building and installing packages in the Astropy ecosystem")
+  (home-page "https://extension-helpers.readthedocs.io")
+  (synopsis "Astropy ecosystem utilities for building and installing packages")
   (description
-    "The extension-helpers package includes many build, installation, and
-documentation-related tools used by the Astropy project.")
+    "The extension-helpers package includes convenience helpers to assist with
+building Python packages with compiled C/Cython extensions.  It is developed by
+the Astropy project but is intended to be general and usable by any Python
+package.")
   (license license:bsd-3)))
 
 (define-public python-extras
-- 
2.39.2

